lib/hanami/application.rb

```diff
@@ -222,54 +219,48 @@ def prepare_base_load_path
         $LOAD_PATH.unshift base_path unless $LOAD_PATH.include?(base_path)
       end
 
+      # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength
       def prepare_container
-        define_container.tap do |container|
-          configure_container container
-        end
-      end
-
-      def prepare_deps_module
-        define_deps_module
-      end
-
-      def define_container
-        require "#{application_name}/container"
-        namespace.const_get :Container
-      rescue LoadError, NameError
-        namespace.const_set :Container, Class.new(Dry::System::Container)
-      end
+        container =
+          begin
+            require "#{application_name}/container"
+            namespace.const_get :Container
+          rescue LoadError, NameError
+            namespace.const_set :Container, Class.new(Dry::System::Container)
```

Comment:
Could we use dry-core's class builder so that the anonymouse class is a bit nicer to work with in the IRB? ie:

```ruby
irb(main):010:0> Class.new(Dry::System::Container)
=> #<Class:0x00007faae3b7c7b8>
irb(main):011:0> Dry::Core::ClassBuilder.new(parent: Dry::System::Container, name: "MyContainer").()
=> MyContainer
```

Reply:
Happy to - I plan to polish all the container setup for the alpha after this one :)

Reply:
FWIW I think it already gets a nice name after it's assigned the the const, anyway (but I'll double check all of this later in this month).